fl4health/clients/flash_client.py

```diff
@@ -0,0 +1,139 @@
+                local_step += 1
+
+            metrics = self.train_metric_manager.compute()
+            loss_dict = self.train_loss_meter.compute().as_dict()
+            current_loss = loss_dict.get("backward", 0.0)
+
+            # Early stopping check
+            if self.gamma is not None and abs(previous_loss - current_loss) < self.gamma / (local_step + 1):
+                log(
+                    INFO, f"Early stopping at epoch {local_epoch} with loss change {abs(previous_loss - current_loss)}"
+                )
+                break
+
+            previous_loss = current_loss
+
+            self._handle_logging(loss_dict, metrics, current_round=current_round, current_epoch=local_epoch)
+            self._handle_reporting(loss_dict, metrics, current_round=current_round)
+
+        return loss_dict, metrics
+
+    def train_by_steps(
+        self, steps: int, current_round: Optional[int] = None
+    ) -> Tuple[Dict[str, float], Dict[str, Scalar]]:
+        self.model.train()
+
+        # Pass loader to iterator so we can step through train loader
+        train_iterator = iter(self.train_loader)
+
+        self.train_loss_meter.clear()
+        self.train_metric_manager.clear()
+        previous_loss = float("inf")
+        for step in range(steps):
+            try:
+                input, target = next(train_iterator)
+            except StopIteration:
+                train_iterator = iter(self.train_loader)
+                input, target = next(train_iterator)
+
+            if self.is_empty_batch(input):
+                log(INFO, "Empty batch generated by data loader. Skipping step.")
+                continue
+
+            input, target = self._move_input_data_to_device(input), target.to(self.device)
+            losses, preds = self.train_step(input, target)
+            self.train_loss_meter.update(losses)
+            self.train_metric_manager.update(preds, target)
+            self.update_after_step(step)
+            self.total_steps += 1
+
+            loss_dict = self.train_loss_meter.compute().as_dict()
+            current_loss = loss_dict.get("backward", 0.0)
+
+            # Early stopping check
+            if self.gamma is not None and abs(previous_loss - current_loss) < self.gamma / (step + 1):
+                log(INFO, f"Early stopping at step {step} with loss change {abs(previous_loss - current_loss)}")
+                break
+
+            previous_loss = current_loss
+
+        loss_dict = self.train_loss_meter.compute().as_dict()
+        metrics = self.train_metric_manager.compute()
+
+        self._handle_logging(loss_dict, metrics, current_round=current_round)
+        self._handle_reporting(loss_dict, metrics, current_round=current_round)
+
+        return loss_dict, metrics
+
+    def setup_client(self, config: Config) -> None:
+        super().setup_client(config)
+        self.gamma = self.narrow_config_type(config, "gamma", float) if "gamma" in config else 0.04
```

Comment:
This will imply that the gamma always has a value (i.e. the user doesn't have the option to turn off early stopping. I'd say that we could just do
```
if "gamma" in config:
    self.gamma = self.narrow_config_type(config, "gamma", float)
```
and leave it none if the user has not specified the threshold.

Reply:
done!